lms/djangoapps/certificates/models.py

```diff
@@ -309,6 +315,61 @@ def __unicode__(self):
                ("regenerated" if self.is_regeneration else "generated", self.generated_by, self.created, self.course_id)
 
 
+class CertificateInvalidation(TimeStampedModel):
+    """
+    Model for storing Certificate Invalidation.
+    """
+    generated_certificate = models.ForeignKey(GeneratedCertificate)
+    invalidated_by = models.ForeignKey(User)
+    notes = models.TextField(default=None, null=True)
+    active = models.BooleanField(default=True)
```

Comment:
@asadiqbal08 I added `active` field so that when `CertificateInvalidation` needs to be deleted we could set `active=False` instead of removing it from the db that is why `default` for `active` is `True` and it will be set `False` when trying to delete this record.